### PR TITLE
Enable carbonated 02

### PR DESCRIPTION
**Pull request by AlexanderOreshko**
Moving Carbonated specific EMotionFX

### CODE REVIEW

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/Component/EntityId.h>
+
+namespace EMotionFX
+{
+    namespace Integration
+    {
+#if defined(CARBONATED)
```
Comment (alexeykostin) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h]: Is there any reason to keep `#if defined(CARBONATED)` nested?

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+         */
+        class ApplyMotionSetComponentRequests
+            : public AZ::ComponentBus
+        {
+        public:
+
+            static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+
+            // applies the defined motion set to this Entity's AnimGraphInstance
+#if defined(CARBONATED)
```
Comment (alexeykostin) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h]: Is there any reason to keep #if defined(CARBONATED) nested?

**Review by vmednonogov — APPROVED** [review]
LGTM
Please fix Alexey's comments first

**Review by alexeykostin — COMMENTED** [review]
It looks like also changes in 

```
Integration\Components\ApplyMotionSetComponent.cpp
Integration\Editor\Components\EditorApplyMotionSetComponent.cpp
Integration\System\AnimationModule.cpp
```
should be moved.

**Pull request closed without merging**